src/repo/git_test.go

```diff
@@ -53,7 +53,7 @@ func TestClone(t *testing.T) {
 	_, err = gitComponent.Clone(repo, "", "main", "1234")
 	assert.Equal(t, domain.ErrGitBranchAndShaMutuallyExclusive, err, "must give an error if branch AND sha passed")
 
-	firstGitRepo, err := gitComponent.Clone(repo, "dummy-access-token", "master", "")
+	firstGitRepo, err := gitComponent.Clone(repo, "dummy-access-token", "master", "", &domain.ProgressBuf{})
```

Comment:
I think it will fail in tests, look how it's created in githubWebhook, it has a buffer field, here it's nil.
that's why I suggest to create a function NewProgressBuf that will initialise all the necessary fields